Reorganize packaging docs to match the workflow.

TraceyC77 opened the pull request:
## Description

Add useful links. Clarify wording. Move text blocks to where they make sense in the workflow.

### Submitter Checklist

- [x] Squashed commits with `git rebase -i` (if needed)

TraceyC77 commented:
I would like to note that this is the first time I've tried to use an include with Docusaurus. If it works right, the danger box should appear on the index.mdx page as well as prepare-for-packaging.mdx. If this doesn't work as expected, I'll update things to remove the include and just use the text.

EbonJaeger commented on docs/packaging/creating-a-new-package.md:
```diff
@@ -83,11 +118,13 @@ install    : |
 ## Build the Package
 
-Build the package:
+Build the package (`tree` uses `make`, other packages may use another command):
```
What build system the software being packaged uses is irrelevant; to build a package it will always be `make` (until we switch to `go-task`).

EbonJaeger replied:
Unless you mean like `make local`? Either way, it might give readers the impression that the package's build system matters.

TraceyC77 replied:
Ah, thanks for clarifying that. I had conflated the invocation in package.yml. with the command line build command. Fixor.

EbonJaeger commented on docs/packaging/submitting-a-package.md:
````diff
@@ -1,59 +1,90 @@
+
+It is a good idea to remove the package you built, and return to the repository version, once you are done testing. To do this, run the following command on the main application. Example:
+
+```bash
+sudo eopkg it --reinstall nano
 ```
````
```suggestion
```bash
sudo eopkg it --reinstall tree
```
```

TraceyC77 replied:
Sure, since tree was mentioned in the doc about your first package update.

EbonJaeger commented on docs/packaging/submitting-a-package.md:
```diff
@@ -1,59 +1,90 @@
+
+### Returning to the Repository Version (optional but recommended)
+
+It is a good idea to remove the package you built, and return to the repository version, once you are done testing. To do this, run the following command on the main application. Example:
```
Main application?

EbonJaeger commented on docs/packaging/submitting-a-package.md:
```diff
@@ -63,45 +94,78 @@ For files that must be removed, you must do so using git: `git rm someFile`
+For more information on suitable commit messages, please check the [tooling central documentation](https://github.com/solus-project/tooling-central/blob/master/README.rst#using-git).
 
-- If you want to link this patch to an issue on the Dev Tracker, simply mention it in your commit message: `The inclusion of <somepackage> fixes #1234`
+- If you want to link this pull request to an existing issue, simply mention it in your commit message: `The inclusion of <somepackage> fixes #1234`
 - If you need a change to depend on another change, mention it in the commit message too: `Depends on #123`
```
Since issues are in a different repository (and organization), it's best to put in the full URL. You can probably get away with just `getsolus/packages#1234` (or something like that; I don't quite remember the correct regex off the top of my head), but just copying the full URL is likely faster and easier.

TraceyC77 replied:
Good point. Fixed.

EbonJaeger commented on docs/packaging/submitting-a-package.md:
```diff
@@ -128,6 +192,6 @@ Pushing changes is not possible unless you have maintainer access. The same is a
 
```
I can't put a comment on the line I want to because it's not technically a part of this PR, but there is a missing period after the link to the build status page.

TraceyC77 replied:
Fixed

EbonJaeger commented on docs/packaging/submitting-a-package.md:
```diff
@@ -128,6 +192,6 @@ Pushing changes is not possible unless you have maintainer access. The same is a
 To request maintainer rights for a repository, it is expected that some level of contribution/maintenance has already happened by way of testing/patching, and there is reasonable trust demonstrated to "hand the keys" over to a repository.
 
-Currently, the request mechanism is [contact Solus Staff on Matrix](/docs/user/contributing/getting-involved#matrix-chat)
+Currently, the request mechanism is [to contact Solus Staff on matrix in the Packaging room](/docs/user/contributing/getting-involved#matrix-chat).
```
```suggestion
Currently, the request mechanism is [to contact Solus Staff on Matrix in the Packaging room](/docs/user/contributing/getting-involved#matrix-chat).
```

EbonJaeger commented on docs/packaging/updating-an-existing-package.md:
````diff
@@ -1,13 +1,45 @@
-The instructions below assume you have cloned the package's repository using the https link provided on the respective git page, [for example nano](https://github.com/solus-packages/nano). The process for submitting an updated package is the same as if it is a new package. Follow the steps [here](/docs/packaging/submitting-a-package).
+The instructions below assume you have cloned the package's repository. The easiest way to do this is (substitute the actual package name):
+
+```bash
+git make package-name.clone
+```
````
```suggestion
```bash
make package-name.clone
```
```

EbonJaeger commented on docs/packaging/updating-an-existing-package.md:
```diff
@@ -1,13 +1,45 @@
+
+## Update Your GitHub Fork
+
+If you have a fork of the package in GitHub, from previous updates, log into GitHub. Make sure you're looking at the master branch & that your fork is up to date with the repo it was forked from. If your fork indicates it is behind, use the "Sync fork" button to bring it up to date.
```
The reader would still have to pull to their local clone of the fork, right? That should probably be added here.

TraceyC77 replied:
Good point. Added the commands for it.

EbonJaeger commented on docs/packaging/updating-an-existing-package.md:
```diff
@@ -1,13 +1,45 @@
+
+## Switch to a New Git Branch
+
+It's always a good idea to switch to a new git branch before beginning packaging work.   This will allow you to more easily separate your work from an new changes made to the package repository, which will allow you to more easily rebase any changes if needed.
```
```suggestion
It's always a good idea to switch to a new git branch before beginning packaging work. This will allow you to more easily separate your work from an new changes made to the package repository, which will allow you to more easily rebase any changes if needed.
```

EbonJaeger commented on docs/packaging/updating-an-existing-package.md:
````diff
@@ -43,3 +75,13 @@ After bumping or updating the package, build it:
 make
 ```
+
+## Building a package
+
+After bumping or updating the package, build it using the appropriate command. Usually this is:
+
+```bash
+make
+```
````
Section above says pretty much the same thing.

TraceyC77 replied:
This was an oopsie from the rebase. I've removed the redundant text.

EbonJaeger commented on docs/packaging/package.yml.md:
```diff
@@ -3,15 +3,15 @@ title: Package YAML
+# `package.yml`
 
-All packages are generated from a single build file, which provides all of the required metadata for the package manager, plus the packaging steps involved to produce a package. This follows the YAML specification.
+Each package is generated from a single build file `package.yml`. This provides all of the required metadata for the package manager, plus the packaging steps involved to produce a package. This file follows the YAML specification.
```
```suggestion
Each package is generated from a single build file named `package.yml`. This provides all of the required metadata for the package manager, plus the packaging steps involved to produce a package. This file follows the YAML specification.
```

EbonJaeger commented on docs/packaging/index.mdx:
```diff
@@ -0,0 +1,34 @@
+
+1. [Update your development environment](update-dev-environment.md)
+2. Prepare the package directory and build the package
+* For a new package that does not yet exist in the repos see [Creating a New Package](creating-a-new-package.md)
+* For updating a package that already is in the repos see [Updating an Existing Package](updating-an-existing-package.md)
```
```suggestion
  * For a new package that does not yet exist in the repos see [Creating a New Package](creating-a-new-package.md)
  * For updating a package that already is in the repos see [Updating an Existing Package](updating-an-existing-package.md)
```

EbonJaeger reviewed (changes requested):
I went through setting up the packaging environment and creating a new package as outlined in Creating A New Package, only directly copy/pasting the given commands. Much, much easier to follow now than it used to be!

TraceyC77 commented on docs/packaging/prepare-for-packaging.mdx:
```diff
@@ -1,19 +1,18 @@
+<MigrationWarning/>
 
 # Prepare for Packaging
 
```
Good point, that was an oversight. I've added a blurb at the top of this document.

TraceyC77 commented on docs/packaging/prepare-for-packaging.mdx:
```diff
@@ -40,24 +39,23 @@ sudo eopkg it -c system.devel
 ## Setting up a GitHub account
 
 The solus source repositories for the package repository currently reside on [github.com/solus-packages](https://github.com/solus-packages). You will need a GitHub account to submit patches and file issues. You can create a GitHub account [here](https://github.com/signup).
 
```
Good idea. I've added a paragraph about this in the section about setting up a GitHub account.

TraceyC77 commented on docs/packaging/prepare-for-packaging.mdx:
````diff
@@ -84,11 +82,16 @@ mkdir ~/solus-builds
+
+Change into the directory you will keep the folders for the packages you will be buildingand clone the common repository with git. For instance if you have a package directory `/home/user/packaging` you would run:
 
-Next you need to set up `common`, a set of scripts that enables you to manage, build, check, and publish packages.
+```bash
+cd /home/user/packaging
+git clone https://dev.getsol.us/source/common.git
+```
 
-Run the following commands to clone the `common` repository and create the required symlinks. Do this inside `solus-builds`:
+Next you need to set up a few symlinks. Do this from the same directory you executed the `git` command:
````
I've fixed the directory naming to be consistent and fixed the URL.

TraceyC77 commented:
Thanks for the thorough review. I believe I've addressed everything you highlighted. Let me know if there are any other improvements you see can be made.

ermo commented on docs/packaging/creating-a-new-package.md:
```diff
@@ -1,37 +1,68 @@
+
+## Initialize Git Repo
+
+Initialize the folder as a git repository and switch to a new branch.  This will allow you to more easily separate your work from an new changes made to the package repository, which will allow you to more easily rebase any changes if needed.
```
`an new changes` -> `any new changes`?

TraceyC77 replied:
Good catch. Fixed.

ermo commented on docs/packaging/package.yml.md:
```diff
@@ -3,15 +3,15 @@ title: Package YAML
+# `package.yml`
 
-All packages are generated from a single build file, which provides all of the required metadata for the package manager, plus the packaging steps involved to produce a package. This follows the YAML specification.
+Each package is generated from a single build file named `package.yml`. This provides all of the required metadata for the package manager, plus the packaging steps involved to produce a package. This file follows the YAML specification.
```
Consider explicitly using the word 'recipe' here rather than 'file'? This meshes well with the use of "packaging steps".

TraceyC77 replied:
I disagree with this. The word "file" exactly describes the thing being referenced (the package.yml file). The word recipe is ambiguous and would be confusing.

I've personally only ever seen a yaml file called a recipe in the context of chef software, and some previous conversations in chat among Solus staff. I think it would be confusing to new packagers without adding value. I'm willing to listen to what Evan and other reviewers have to say.

TraceyC77 replied:
I came up with wording that I think is a good compromise, which introduces the concept of this file being a recipe.

ermo commented on docs/packaging/package.yml.md:
```diff
@@ -3,15 +3,15 @@ title: Package YAML
 All `package.yml` files **must** be valid YAML.
 
-The file is organised into a key→value hierarchy. The `ypkg` tool parses a `package.yml` file to build the corresponding package in a declarative manner, so most of the keys are simple strings, lists or nested key→value pairs. A special case consists in the packaging steps, which are scripts.
+The file is organised into a key→value hierarchy. The `ypkg` tool parses a `package.yml` file to build the corresponding package in a declarative manner. As such, most of the keys are simple strings, lists or nested key→value pairs. A special case consists in the packaging steps, which are scripts.
```
again, consider using 's/file/recipe'.

ermo commented on docs/packaging/procedures/maintainership.md:
```diff
@@ -33,11 +33,11 @@ Maintainers are free to [update a package](/docs/packaging/updating-an-existing-
 ## Template
 
-Presented here is the `MAINTAINERS.md` file. This file must be provided verbatim alongside the other patch contents, and filled in with the maintainers' personal information. A Matrix contact is optional but recommended, while an email address is mandatory. Similar to the `.solus/packager` file used for packaging, the maintainers listed in `MAINTAINERS.md` must use their real first and last name( ...
+Presented here is the `MAINTAINERS.md` file. This file must be provided verbatim alongside the other patch contents, and filled in with the maintainers' personal information. A [Matrix](/docs/user/contributing/getting-involved.md#matrix-chat) contact is optional but recommended, while an email address is mandatory. Similar to the `.solus/packager` file used for packaging, the maintainers listed i ...
```
Don't we use `~/.config/solus/packager` now?

TraceyC77 replied:
Ah yeah, good point. Rewrote this to indicate this is a legacy file that should be removed. I've moved the bits about what is mandatory over to Prepare for Packaging, with the packager file instructions.

silkeh commented on docs/packaging/submitting-a-package.md:
```diff
@@ -63,45 +94,78 @@ For files that must be removed, you must do so using git: `git rm someFile`
+Now that you have your changed committed, it's time to send it to us for review by submitting a pull request to the GitHub repository.
 
-You will first need to fork the repository either from the GitHub web interface. Or by using the `gh` cli tool from the `github-cli` package.
+You will first need to fork the repository. You can do this either from the GitHub web interface, or by using the [`gh` cli tool](https://cli.github.com/manual/gh_repo_fork) from the `github-cli` package.
```
This doesn't work the same for package updates and new packages. Additionally, it would simplify the process if we move this to the beginning, so: 1) fork, 2) clone your fork, 3) do updates, 4) commit, 5) push, 6) create PR. No fiddling with the remote.

TraceyC77 replied:
Agreed. I'm re-doing the create and update docs to re-order the steps and put forking first.

silkeh commented on docs/packaging/updating-an-existing-package.md:
````diff
@@ -1,23 +1,62 @@
+
+```bash
+cd /home/solus-builds/package-name
+git pull
````
```suggestion
git switch master
git pull
```

TraceyC77 replied:
Added.

silkeh reviewed (commented):
Great work!

The workflows for a new package and package update converge at a point that doesn't really work (this was already the case). There is also some nonlinearity when dealing with forks/etc. Maybe it would be clearer to actually document the *entire* process on the two respective pages with the Git basics split into a separate article, so the workflows would look roughly as follows:

- Update existing package:
  1. Fork a repo or update your fork. This means that the starting point is always the same, no difference between the first time and any following update.
  1. Clone repo
  1. Create branch
  1. Bumping/updating/building
  1. Commit (the template should be different between update and new package)
  1. Create PR with a note to a *Working with Git* page with general rework instructions/etc.
- Creating a new package:
  1. Fork the getsolus/packages repo or update your fork.
  1. Clone repo
  1. Create branch
  1. Create a new directory.
  1. Create package/maintainers/build/etc
  1. Commit
  1. Create PR with a note to a *Working with Git* page with general rework instructions/etc.

The goal here would be to document the entire process on a single page with simple and to the point instructions. The current setup where both pages converge when submitting the update requires the 'submitting a package' page to accommodate both, making it more complex and harder to follow.

TraceyC77 commented on docs/packaging/submitting-a-package.md:
```diff
@@ -1,59 +1,90 @@
-- You can checkout the master branch by running `git checkout master`, and switch back to your branch by running `git checkout my-branch`
-- If there any new changes to the repository whilst you are still working on your branch you can run `git pull origin master --rebase` to pull the changes then manually fixup any conflicts.
+- If necessary, you can checkout the master branch by running `git switch master`, and switch back to your branch by running `git switch my-branch`
+- If there any new changes to the repository whilst you are still working on your branch you can run `git pull origin master --rebase` to pull the changes then manually fixup any conflicts. You can check your branch against master with `git diff master..my-branch`.
```
Agreed, updated the wording.

silkeh commented on docs/packaging/creating-a-new-package.md:
```diff
@@ -104,4 +170,40 @@ Once the build completes, your directory should now include the following files:
+
+[Add / remove files as necessary to the commit](git-basics.md). Then, **check your branch**.
+
+Run `git status`. Make sure all the files you changed are staged, and that there are no untracked files. When all it well, run `git commit`.
```
The `##` are stripped by default from the commit message :sweat: 

```suggestion
Run `git status`. Make sure all the files you changed are staged, and that there are no untracked files. When all is well, run `git commit --cleanup=scissors`.
```

TraceyC77 replied:
After discussion in Matrix, I'll change the recommended template to use **bolding** instead of the ##. This avoids having to use git flags, and keeps it simpler.

silkeh reviewed (changes requested):
One minor issue when trying the instructions for a new package, but everything went well otherwise. Great work!

Staudey commented on docs/packaging/procedures/maintainership.md:
```diff
@@ -33,11 +33,11 @@ Maintainers are free to [update a package](/docs/packaging/updating-an-existing-
+## Template for the `MAINTAINERS.md` File
 
-Presented here is the `MAINTAINERS.md` file. This file must be provided verbatim alongside the other patch contents, and filled in with the maintainers' personal information. A Matrix contact is optional but recommended, while an email address is mandatory. Similar to the `.solus/packager` file used for packaging, the maintainers listed in `MAINTAINERS.md` must use their real first and last name( ...
+Presented here is the `MAINTAINERS.md` file. This file must be provided verbatim alongside the other patch contents, and filled in with the maintainers' personal information. A [Matrix](/docs/user/contributing/getting-involved.md#matrix-chat) contact is optional but recommended, while an email address is mandatory. Similar to the `.solus/packager` file used for packaging, the maintainers listed i ...
```
```suggestion
Presented here is the `MAINTAINERS.md` file. This file must be provided verbatim alongside the other patch contents, and filled in with the maintainers' personal information. A [Matrix](/docs/user/contributing/getting-involved#matrix-chat) contact is optional but recommended, while an email address is mandatory. Similar to the `.solus/packager` file used for packaging, the maintainers listed in `MAINTAINERS.md` must use their real first and last name(s) for accountability purposes.
```

For some reason the `.md` causes issues when linking the documents. Only in the second instance for some reason that I can't quite discern atm, but might as well change both since the suffix is not necessary anyway.

TraceyC77 replied:
Fixed all the instances I could find. Thanks for the feedback.

Staudey commented on docs/packaging/prepare-for-packaging.mdx:
```diff
@@ -0,0 +1,153 @@
+## Building Packages
+
+Your system is now set up for package work.
+If you are new to packaging, see [Your First Package](your-first-package.md).
```
```suggestion
If you are new to packaging, see [Your First Package Update](your-first-package-update.md).
```

TraceyC77 replied:
Fixed

TraceyC77 commented on docs/packaging/index.mdx:
```diff
@@ -0,0 +1,34 @@
+2. Prepare the package directory and build the package
+  * For a new package that does not yet exist in the repos see [Creating a New Package](creating-a-new-package.md)
+  * For updating a package that already is in the repos see [Updating an Existing Package](updating-an-existing-package.md)
+3. [Test the Package and Submit the Pull Request for Review](submitting-a-package.md)
```
Fixed

TraceyC77 commented on docs/packaging/updating-an-existing-package.md:
```diff
@@ -36,10 +94,63 @@ Example:
+- If you want to link this pull request to an existing issue, simply mention it in your commit message (use the full URL): `The inclusion of <somepackage> fixes https://github.com/getsolus/packages/issues/123`
+- If you need a change to depend on another change, mention it in the commit message too (use the full URL): `Depends on https://github.com/getsolus/packages/issues/234`
+
+Next, you'll [submit a pull request for review](submitting-a-package.md).
```
Fixed

TraceyC77 commented on docs/packaging/creating-a-new-package.md:
````diff
@@ -1,37 +1,101 @@
+git switch -c my-branch
+```
+
+## Link to the `Makefile` File
````
Agreed, fixed

TraceyC77 commented on docs/packaging/creating-a-new-package.md:
```diff
@@ -1,37 +1,101 @@
+
+## Link to the `Makefile` File
+
+Every package directory needs a file `Makefile`. To create it with the proper contents, run this:
```
Agreed, fixed

TraceyC77 commented on docs/packaging/procedures/maintainership.md:
```diff
@@ -33,11 +33,11 @@ Maintainers are free to [update a package](/docs/packaging/updating-an-existing-
 ### Non-maintainers
 
-If a package is actively maintained, modifications should not occur and the individual should exercise patience when it comes to the updating of software. In some cases, a maintainer may intentionally be holding back a package, or has simply not updated yet. If pertinent, the individual should file a [package update request](/docs/packaging/procedures/request-a-package-update). Alternatively, the ...
+If a package is actively maintained, modifications should not occur and the individual should exercise patience when it comes to updates. In some cases, a maintainer may intentionally be holding back a package, or has simply not updated yet. If pertinent, the individual should file a [package update request](/docs/packaging/procedures/request-a-package-update). Alternatively, the individual can r ...
```
Fixed

TraceyC77 commented on docs/packaging/_markdown-migration-warning.md:
```diff
@@ -0,0 +1,8 @@
+:::danger
+
+We are currently in the process of migrating our source repositories from Phabricator to GitHub. If you have any repos checked out from Phabricator, [migrate them](docs/packaging/procedures/migrate-repo-to-github.md) before proceeding.
```
Fixed

silkeh commented on docs/packaging/creating-a-new-package.md:
```diff
@@ -1,37 +1,101 @@
+
+From within your packaging directory, create a new directory for the new package. Use the package name as the directory name.
+
+Initialize the folder as a git repository and switch to a new branch.  This will allow you to more easily separate your work from any new changes made to the package repository, which will allow you to more easily rebase any changes if needed. Then, create the required Makefile within that directory.
```
```suggestion
Initialize a folder in the packages repository and switch to a new branch.  This will allow you to more easily separate your work from any new changes made to the packages repository, which will allow you to more easily rebase any changes if needed. Then, create the required Makefile within that directory.
```

TraceyC77 replied:
Thanks for clarifying. I've updated this section.

silkeh commented on docs/packaging/creating-a-new-package.md:
```diff
@@ -1,37 +1,101 @@
 mkdir tree
 cd tree
+git init
+git switch -c my-branch
```
These two lines are not needed. The repo is setup above.

TraceyC77 replied:
Just above you recommended
"Initialize a folder in the packages repository and switch to a new branch."

So it would still need to be 
```bash
mkdir tree
cd tree
git switch -c my-branch
```

(I have removed the new branch command that was before this so the user is only doing it once)

TraceyC77 commented:
@FriesischScott, thanks for your review.

> The build directory is sometimes referred to as _build directory_ and sometimes as _packaging directory_. Should probably be unified to avoid confusion.

Good catch. I've standardized the wording to be "packaging directory".

> Creating a new package refers to the monorepo but updating an existing one to the repos under the _solus-packages_ organization. Is this correct for the time being?

I've clarified my understanding on this, and @silkeh has also outlined this. I've corrected this section (which also addresses your comments about "Creating a new package"). 

> ### Amending a New Pull Request
> 
> This section is located _after_ the `push` instruction. This would require `--force` pushing which is a potentially dangerous thing to teach `git` newcomers. Maybe this should instead be where the commit message guidelines are for creating new and updating existing packages? To make sure everything is in order before the PR is made.

This is a valid concern, since git push --force is destructive. I've changed this to recommend doing a regular commit and push.

Silke and I wanted to keep the example commit messages separated between creating / updating since they are different. It would be confusing to have them both in one place, trying to explain both. To make this section clearer, I did put in a bit to remind the user to review their commit message before submitting the PR.

> ### Deleting your branch after a pull request is merged
> 
> Deleting the local branch requires switching back to `master` first, since checked out branches can't be deleted.

Fixed.

> ### Deprecating a package
> 
> The list starts at 0, not sure if that's intended.

Auto numbering woopsie, fixed.

silkeh commented on docs/packaging/git-basics.md:
````diff
@@ -46,4 +46,12 @@ Now that you've tested and reviewed your change, when you're happy with it, it's
+First, change to the package directory, switch to the master branch, then delete your local and the remote branches.
+
+```bash
+cd /home/solus-builds/package
````
```suggestion
cd ~/solus-builds/package
```

TraceyC77 replied:
Good catch. Fixed all occurrences to be be `~/solus-builds/package`

EbonJaeger commented on docs/packaging/procedures/migrate-repos-to-github.md:
````diff
@@ -0,0 +1,42 @@
+```bash
+cd ~/solus-builds/common
+git switch master
+git remote set-url origin https://github.com/getsolus/common
+git remote set-url --push origin git@github.com:getsolus/common.git
````
https URL for one and ssh URL for the other? Why not both be the same?

TraceyC77 replied:
This is what I was given in the Solus Matrix channels. I don't know why its this way. If Silke agrees it should be changed, it can go into his docs update for the monorepo.

EbonJaeger replied:
We can revisit this later. Let's not block on it.

EbonJaeger commented on docs/packaging/creating-a-new-package.md:
````diff
@@ -104,4 +165,40 @@ Once the build completes, your directory should now include the following files:
 ```
 
-All these files _except_ the `*eopkg` file should be included when you submit the package for review.
+All these files _except_ the `*eopkg` file(s) should be included in your pull request. You will remove them after testing the package.
````
"them" is ambiguous here; it could mean the eopkg files, or all of the other files.
```suggestion
All these files _except_ the `*eopkg` file(s) should be included in your pull request. You will remove the `.eopkg` files after testing the package.
```

TraceyC77 replied:
Fixed

EbonJaeger commented on docs/packaging/submitting-a-pull-request.md:
```diff
@@ -0,0 +1,78 @@
+
+:::note
+
+Previously, with Phabricator. we required amending your commit. This is no longer necessary and is not recommended.
```
```suggestion
Previously, with Phabricator, we required amending your commit. This is no longer necessary and is not recommended.
```

TraceyC77 replied:
Fixed

EbonJaeger reviewed (changes requested):
Almost there!

TraceyC77 commented on docs/packaging/updating-an-existing-package.md:
```diff
@@ -1,23 +1,81 @@
+:::note Alternatives to `gh repo clone`
+
+You may use the https link provided on the respective git page, [for example nano](https://github.com/solus-packages/nano) with `git clone`.
+You may use `git make package-name.clone`.
```
In the interest of not confusing people new to packaging with stuff that won't work for them, I'm just going to remove the note. If the go tasks are updated to work with the user's fork, the docs can be updated at that time.

TraceyC77 commented on docs/packaging/prepare-for-packaging.mdx:
```diff
@@ -0,0 +1,153 @@
+
+Next you need to set up `common`. This is a repository with a set of make scripts which enable you to more easily manage, build, check, and publish packages.
+
+Change into the directory you will keep the folders for the packages you will be building and clone the common repository with git. For instance if you have a package directory `/home/user/solus-builds` you would run:
```
Removed the line

TraceyC77 commented on docs/packaging/updating-an-existing-package.md:
````diff
@@ -36,10 +94,63 @@ Example:
-```bash
-make
 ```
+Update foo to 1.2.3
````
Updated

TraceyC77 commented on docs/packaging/updating-an-existing-package.md:
````diff
@@ -36,10 +94,63 @@ Example:
 ```
+Update foo to 1.2.3
+
+**Changelog**
````
We had chosen asterisk bolding here because hash bolding would require special git flags. Otherwise on command line, git commit strips the hashes out, treating them as comments . I've recommend updating the PR template in Silke's monorepo PR https://github.com/getsolus/packages/pull/344
We might decide on a different heading marker (like =) or work around the git issue.

TraceyC77 commented on docs/packaging/updating-an-existing-package.md:
```diff
@@ -36,10 +94,63 @@ Example:
+
+## Commit Your Changes
+
+Check the [changes in your files](git-basics).
+
+[Add / remove files as necessary to the commit](git-basics.md). Then, **check your branch**.
```
Good thought. Added an anchor link.

EbonJaeger reviewed (approved):
LGTM! Really fantastic work. Thank you for taking on this monumental task! :)
